src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Private/JavaScriptMarshalImpl.Bind.cs

```diff
@@ -12,10 +12,10 @@ public static Action<T1> BindJSActionByName<T1>(string functionName)
         {
             const int argsCount = 1;
             var extraBufferLength = 0;
-            var excSig = GetExceptionMeta();
-            var resSig = GetVoidMeta();
-            var arg1Sig = GetArgumentMeta<T1>(ref extraBufferLength);
-            var signature = stackalloc JavaScriptMarshallerArgSig[] { excSig, resSig, arg1Sig };
+            var excSig = GetExceptionSignature();
+            var resSig = GetVoidSignature();
+            var arg1Sig = GetArgumentSignature<T1>(ref extraBufferLength);
```

Comment:
It may be better to make this and the others static readonly fields so that we don't have to perform a function call. The generic ones could be fields of a generic type - that may or may not be better (we'd need to find out how it behaves in AOT), since the generics would possibly be resolved at compile/class init time.

Reply:
done, it also solved some pointer/allocation problem because I now allocate and cache unmanaged memory for the signature buffer. Thanks!